### PR TITLE
Addition of a circuit that checks FRI proximity proofs

### PR DESCRIPTION
**Pull request by ThomasPiellard**
* Merkle proofs verification in circuit now use the binary decomposition of the leaf to know the order for hashing (instead of giving a sequence of "proof helpers" before)
* The rollup example has been fixed accordingly, and also each input is constrained now
* Addition of the fri verification circuit in `std/commitments/fri/`. The verification circuit checks radix-2 fri proximity proof as formatted in gnark-crypto. In particular, Fiat Shamir in gnark-crypto, when used with fri, has been modified so that the challenges seen as string outside a circuit and as frontend.Variable inside of the circuit match.

### CODE REVIEW

File: examples/rollup/circuit.go [hunk 1]
```diff
@@ -108,12 +104,28 @@ func (circuit *Circuit) postInit(api frontend.API) error {
 		circuit.Transfers[i].SenderPubKey = circuit.PublicKeysSender[i]
 		circuit.Transfers[i].ReceiverPubKey = circuit.PublicKeysReceiver[i]
 
+		// allocate the slices for the Merkle proofs
+		// circuit.allocateSlicesMerkleProofs()
```
Comment (gbotrel) [inline examples/rollup/circuit.go:108]: why is this commented?

File: examples/rollup/circuit_test.go [hunk 2]
```diff
@@ -195,6 +213,7 @@ func TestCircuitUpdateAccount(t *testing.T) {
 	assert := test.NewAssert(t)
 
 	var updateAccountCircuit circuitUpdateAccount
+	(*Circuit)(&updateAccountCircuit).allocateSlicesMerkleProofs()
```
Comment (gbotrel) [inline examples/rollup/circuit_test.go:216]: why the cast?

**Review by gbotrel — CHANGES_REQUESTED** [review]
need to fix staticcheck errors + would have been nicer to have 3 separate PRs :-)

File: examples/rollup/circuit_test.go (hunk 2 above)
Comment (ThomasPiellard) [reply examples/rollup/circuit_test.go:216]: type `circuitUpdateAccount` inherits the fields but not the methods of `Circuit`

**Comment by gbotrel** [comment]
👍  just staticcheck linter to fix and is good to merge

**Comment by gbotrel** [comment]
@ThomasPiellard did a forge push on that one, gnark-crypto update was done in another branch 👍

**Review by gbotrel — APPROVED** [review]

**Pull request merged**